### PR TITLE
[Shortcircuit] Add voltage value in VoltageRange

### PR DESCRIPTION
**Pull request by colinepiloquet**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

- `ShortCircuitParameters` is in version 1.2;
- `FaultParameters` is in version 1.2;
- `VoltageRange` contains the following data:
  - low voltage;
  - high voltage;
  - range coefficient.

**What is the new behavior (if this is a feature change)?**
- `ShortCircuitParameters` is in version **1.3**;
- `FaultParameters` is in version **1.3**;
- `VoltageRange` contains the following data:
  - low voltage;
  - high voltage;
  - range coefficient;
  - **voltage**.

The new `voltage` field in `VoltageRange` is optional and indicates which voltage should be used for the equipments which nominal voltages are between the low and the high voltage of the range.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [X] No

**Other information**:

### CODE REVIEW

**Comment by colinepiloquet** [comment]
@annetill, @olperr1 a suggestion for voltage in the voltageRange. If you validate, I also add it in FaultParameters + finish coding.

**Comment by olperr1** [comment]
> @annetill, @olperr1 a suggestion for voltage in the voltageRange. If you validate, I also add it in FaultParameters + finish coding.

It's OK for me. It's maybe the clearest thing we can have.

With this method, we can no more define a coefficient for [0 ; 230] (range which has several values for `voltage`) in a single range, but I think it's for the best since it will also highlight problematic cases where the coefficient has several values for a single value of `voltage`.

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: shortcircuit-api/src/main/java/com/powsybl/shortcircuit/json/VoltageRangeDeserializer.java [hunk 1]
```diff
@@ -43,10 +44,14 @@ public VoltageRange deserialize(JsonParser parser, DeserializationContext contex
                     parser.nextToken();
                     coefficient = parser.readValueAs(Double.class);
                 }
+                case "voltage" -> {
```
[the remark this answers is not in the document]
Comment (olperr1) [reply shortcircuit-api/src/main/java/com/powsybl/shortcircuit/json/VoltageRangeDeserializer.java:47]: You could use the following methods :
- `JsonUtil.setSourceVersion(…)`: to store the version number in the deserialization context (in `FaultParametersDeserializer` and in `ShortCircuitParametersDeserializer`);
- `Json.getSourceVersion(…)`: to retrieve it (in `VoltageRangeDeserializer`).

[the remark this answers is not in the document]
Comment (colinepiloquet) [reply shortcircuit-api/src/main/java/com/powsybl/shortcircuit/json/VoltageRangeDeserializer.java:47]: This is really great!! I didn't know it existed, I should have used it for other deserializers (it goes into my todo list)

**Review by olperr1 — APPROVED** [review]

**Pull request merged**